table/tables/index.go

```diff
@@ -149,7 +150,16 @@ func (c *index) truncateIndexValuesIfNeeded(indexedValues []types.Datum) []types
 				// truncate value and limit its length
 				v.SetBytes(colValue[:ic.Length])
 			}
+			// Restore the orginal datum kind. Otherwise the admin check table will fail.
```

Comment:
How about this:
use below instead of line #147 
```go
					if v.Kind() == types.KindString {
						v.SetString(truncateStr)
					} else {
						v.SetBytes([]byte(truncateStr))
					}
```
then we can remove line #154 ~ #160
and also can remove `adjustDatumKind ` function

Reply:
Cannot understand this comment.

Reply:
adjustDatumKind is used for the old data. And we use v.SetString here can save a memory allocation. @crazycs520